gomint-api/src/main/java/io/gomint/plugin/Plugin.java

```diff
@@ -206,4 +215,31 @@ public GoMint server() {
         return this.server;
     }
 
+    /**
+     * Get loaded worlds (by {@linkplain World#folder()} name) set based on {@code worlds.yml} configuration file in
+     * this plugin's data folder.
+     * Returns {@code null} if plugin should run in all worlds.
+     * Returned set is actively updated on world (un-)load (if not {@code null}).
+     *
+     * @return The set of worlds (by {@linkplain World#folder()} name) the given plugin should be active in. Returns
+     * {@code null} if plugin should run in all worlds.
+     * @see #activeInWorld(World)
+     */
+    @Nullable
+    public ConcurrentHashMap.KeySetView<String, Boolean> activeWorlds() {
```

Comment:
What is the general idea behind this? Why does a plugin have active worlds? How is this handled? Should plugins be restricted to those worlds (block setting, world getting, entity manipulations etc.) and unloaded/loaded for each of that worlds?

Reply:
The idea is that a plugin *can* and *should* use this to restrict itself to certain worlds. The idea is to use this Set to register its listeners easily to restrict those to certain worlds. See also the automatically updating in the comment at registerListener

Reply:
I see, then maybe this PR should be coupled with a documentation document in https://github.com/gomint/gomint-docs stating that system and its goals.

Reply:
Will do